### ISSUE TITLE
Filecoin Archiving through Buckets: Bucket Cleanup

### ISSUE THREAD

**Issue by jimmylee** [textileio/textile#369]
Hey 👋 

I think that there should be a way to **delete buckets** after a Filecoin storage deal has been made. Tagging @asutula @andrewxhill and @pooja for further discussion.

* After the deal is successful, I have no need for the bucket anymore.
* Maybe we can keep track of some other type of reference so I can still refer to a successful deal.
* Deleting buckets after deals are successful can free up a lot of extra unused storage, I can do this for the users of Slate today.

Thanks everyone 🧇

**Comment by jimmylee** [comment]
I spoke to Aaron and his `info` method in Powergate will cover this.

**Comment by asutula** [comment]
If we assume that the only powergate storage of data from hub are bucket archives, then yes, powergate will hold the records of storage, and we know that those records correspond to bucket archives. If, however, there become other ways of storing data in powergate via hub, or if we want to persist any metadata about the deleted/archived bukcet (name, for example), then it will be necessary to store that data in hub. Going to reopen this because I think that second secenario is more likely to play out than the first.

**Comment by jimmylee** [comment]
If I understand correctly, `bucket.archive` records are stored in `Powergate` and that means I don't have to keep the bucket? Just making sure.

**Comment by asutula** [comment]
Correct. In theory, you could recover the data from Filecoin directly, no hub or powergate needed. In practice, that would be much easier if hub hung onto a record of your archived bucket (like imagine a "trash bin" or "archived" folder) and could provide a workflow to re-create the bucket from the archive for you.

**Comment by asutula** [comment]
If hub doesn't provide that bookkeeping, users would have to keep track of all the Filecoin details themselves (data cid, wallet addresses, etc)

**Comment by jimmylee** [comment]
I would love a retrieval method in the hub, where:

* you could call something like `buckets.retrieve(dealCID, minerId)`
* It checks to make sure the deal is under the buckets maximum storage capacity.
* it creates a new bucket with the contents of the retrieval deal.
